addons/sale/report/sale_report.py

```diff
@@ -67,6 +68,7 @@ def _query(self, with_clause='', fields={}, groupby='', from_clause=''):
             t.uom_id as product_uom,
             CASE WHEN l.product_id IS NOT NULL THEN sum(l.product_uom_qty / u.factor * u2.factor) ELSE 0 END as product_uom_qty,
             CASE WHEN l.product_id IS NOT NULL THEN sum(l.qty_delivered / u.factor * u2.factor) ELSE 0 END as qty_delivered,
+            CASE WHEN l.product_id IS NOT NULL THEN (product_uom_qty - qty_delivered)/ u.factor * u2.factor ELSE 0 END as qty_to_deliver,
```

Comment:
```sql
SUM((l.product_uom_qty - l.qty_delivered) / u.factor * u2.factor) ...
```
to keep the grouping logic like the other lines (and that way, you don't have to modify the `groupby_` part).